### PR TITLE
Nginx Proxy

### PR DESCRIPTION
**Pull request by tonybenoy**
I've implemented nginx installation along with your setup. Not the most perfect implementation.

### CODE REVIEW

**Comment by Yenthe666** [comment]
Hi @tonybenoy,

Thanks a lot for your PR and your interest in the installation script!
Why don't you add the Nginx part in the existing install script and create a new configuration variable at the top `configure_nginx = False`? If the user manually sets it to true you can then install & configure Nginx. This has a few benefits:
- Clean integration in one script
- Easy detection based on `True` or `False`
- Easily configurable for end user

What do you think?

**Comment by tonybenoy** [comment]
Yes. I thought about that but then thought it would clutter your existing script.

**Comment by tonybenoy** [comment]
@Yenthe666  I've changed the script as you said do take a look

**Comment by Yenthe666** [comment]
@tonybenoy neat, thanks! I'd set the variable to `False` by default for installing Nginx though.
I'll try to test this script as soon as possible. Do send me a reminder if a few days if you got no feedback

**Comment by tonybenoy** [comment]
@Yenthe666  Were you able to test it out?

**Comment by Yenthe666** [comment]
@tonybenoy sorry I haven't yet, was on holiday. Can you send me a ping here again after reading this? I'll keep it in my active list on Github as a reminder then. Will try to check ASAP.

P.S: I think we should also look at the PR from @chris001 which covers this and loads of other functionalities. & would only merge it in master I think.

**Comment by tonybenoy** [comment]
Ah. I honestly didn't go through @chris001 's PR. I'll try and look into the changes he has made and see how this would work into that.

**Comment by Yenthe666** [comment]
I'm kind off waiting to hear back from him, but didn't hear anything in a while. I'd love that one in as it contains so many functionalities.

**Review by CasVissers-360ERP — APPROVED** [review]
Functional test, works fine.

File: odoo_install.sh [hunk 1]
```diff
@@ -148,12 +151,15 @@ sudo chown -R $OE_USER:$OE_USER $OE_HOME/*
 
 echo -e "* Create server config file"
 
+
 sudo touch /etc/${OE_CONFIG}.conf
 echo -e "* Creating server config file"
 sudo su root -c "printf '[options] \n; This is the password that allows database operations:\n' >> /etc/${OE_CONFIG}.conf"
 sudo su root -c "printf 'admin_passwd = ${OE_SUPERADMIN}\n' >> /etc/${OE_CONFIG}.conf"
 sudo su root -c "printf 'xmlrpc_port = ${OE_PORT}\n' >> /etc/${OE_CONFIG}.conf"
+sudo su root -c "printf 'proxy_mode = True\n' >> /etc/${OE_CONFIG}.conf"
```
Comment (Yenthe666) [inline odoo_install.sh]: This should also have an if check to make sure `INSTALL_NGINX` is true

File: odoo_install.sh [hunk 2]
```diff
@@ -29,14 +29,17 @@ OE_PORT="8069"
 OE_VERSION="11.0"
 # Set this to True if you want to install Odoo 11 Enterprise!
 IS_ENTERPRISE="False"
+# Set this to True if you want to install Nginx!
+INSTALL_NGINX="True"
```
Comment (Yenthe666) [inline odoo_install.sh]: Perhaps it is safer to default to `False`?

**Review by Yenthe666 — CHANGES_REQUESTED** [review]

**Comment by Yenthe666** [comment]
@CasVissers thanks for the test. On which Ubuntu version did you test? Did you also add SSL certs or just plain HTTP?

File: odoo_install.sh (hunk 2 above)
Comment (tonybenoy) [reply odoo_install.sh]: I kind of always install it along with nginx hence it was set to true

**Comment by tonybenoy** [comment]
@yenthe666 done with the changes you requested.

**Comment by CasVissers-360ERP** [comment]
@Yenthe666 I used plain HTTP for the test. It would be nice to have an SSL option. In this case we should define the locations of the SSL key and cert files first. 

But the install script with HTTP only already offers a quicker deploy.

**Comment by Yenthe666** [comment]
Thank you guys! I haven't heard back from @chris001 so I guess I just have to test this. If all is well I think we should merge this in. Remind me if you don't hear back in like two weeks.

**Comment by chris001** [comment]
Hi guys, sorry to be away for some time, life was keeping me like -______-
I'll do my best to reconcile the merge warnings with my PR this week.
Definitely don't be shy and feel free to remind me as often as you like I don't mind at all, this fixing the merging issue absolutely must get done, the script is so much more extremely helpful and powerful and secure to deploy and keep Odoo updated, and even allows for automated testing of deployment here in github. :)

**Comment by Yenthe666** [comment]
Exactly, I'm happy to see you're back and I hope you're doing fine @chris001

**Comment by Yenthe666** [comment]
@tonybenoy can you retarget all of this to https://github.com/Yenthe666/InstallScript/tree/12.0-beta ? I'll test it and merge it in within 48 hours if all is well.

**Comment by tonybenoy** [comment]
Hey @Yenthe666 . Sorry for not responding. I was not available for the time. I'm trying to make some time to make all the changes that you suggested. I did try with the minor name changes and got some issues. So I'll do some more proper testing and get back to you :)

**Pull request closed without merging**